lua/orgmode/objects/date.lua

```diff
@@ -481,7 +481,8 @@ function Date:end_of(span)
   end
 
   if span == 'month' then
-    return self:add({ month = 1 }):start_of('month'):adjust('-1d'):end_of('day')
+    local date = os.date('*t', self.timestamp)
+    return self:set({ day = Date._days_of_month(date) }):end_of('day')
```

Comment:
Use `self` when it is available in the method. We can also move these new methods to be on an instance.
```suggestion
    return self:set({ day = self:_days_of_month(date) }):end_of('day')
```

Reply:
I actually tried it with self, but date here is the table from `os.date`, not `OrgDate`. It get's clearer, when I open my PR, where I cleaned up the type warnings.
What I thought about to make it a local function instead. We can then add a public wrapper for OrgDate. But I would prefer to do this with the next PR.